src/ape/_compat.py

```diff
@@ -0,0 +1,13 @@
+import sys
+
+# We can remove this once we stop supporting python3.7.
+if sys.version_info >= (3, 8):
+    from functools import cached_property  # type: ignore
+    from functools import singledispatchmethod  # type: ignore
```

Comment:
This does break vyper and solidity but I can make sure to release hot-fixes for those.

I think each plugin should manage their own compatibility imports individually to be safe.. Otherwise, they could import from the protect `_compat.py` module and that is okay I think since they are 2nd class plugins